app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -34,35 +40,108 @@ fun ConversationScreen(
     )
 }
 
-@OptIn(ExperimentalMaterial3Api::class)
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalMaterialApi::class)
 @Composable
 private fun ConversationScreen(
     conversationViewState: ConversationViewState,
     onMessageChanged: (TextFieldValue) -> Unit,
     onSendButtonClicked: () -> Unit,
     onBackButtonClick: () -> Unit
 ) {
+    val conversationScreenState = rememberConversationScreenState()
+
     with(conversationViewState) {
-        Scaffold(
-            topBar = { ConversationScreenTopAppBar(conversationName, onBackButtonClick, {}, {}, {}) },
+        MenuModalSheetLayout(
+            sheetState = conversationScreenState.modalBottomSheetState,
+            menuItems = EditMessageMenuItems(
+                editMessageSource = conversationScreenState.editMessageSource,
+                onCopyMessage = conversationScreenState::copyMessage
+            ),
             content = {
-                ConversationScreenContent(
-                    messages = messages,
-                    onMessageChanged = onMessageChanged,
-                    messageText = conversationViewState.messageText,
-                    onSendButtonClicked = onSendButtonClicked
+                Scaffold(
+                    topBar = {
+                        ConversationScreenTopAppBar(
+                            title = conversationName,
+                            onBackButtonClick = onBackButtonClick,
+                            onDropDownClick = {},
+                            onSearchButtonClick = {},
+                            onVideoButtonClick = {}
+                        )
+                    },
+                    snackbarHost = {
+                        SwipeDismissSnackbarHost(
+                            hostState = conversationScreenState.snackBarHostState,
+                            modifier = Modifier.fillMaxWidth()
+                        )
+                    },
+                    content = {
+                        ConversationScreenContent(
+                            messages = messages,
+                            onMessageChanged = onMessageChanged,
+                            messageText = conversationViewState.messageText,
+                            onSendButtonClicked = onSendButtonClicked,
+                            onShowContextMenu = { message -> conversationScreenState.showEditContextMenu(message) }
+                        )
+                    }
                 )
             }
         )
     }
 }
 
+@Composable
+private fun EditMessageMenuItems(
+    editMessageSource: MessageSource?,
+    onCopyMessage: () -> Unit
```

Comment:
can we also add a `onMessageDeleted()` lambda?

Reply:
will add a TODO for now